src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joined = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = separator;
+        joined += this[i];
+      } else if (separator === null) {
+        joined += this[i] + 'null';
```

Comment:
null не треба окремо обробляти якщо null додавати до рядка він теж перетворюється в рядок
```suggestion
      } else if (separator === null) {
        joined = joined + this[i] + separator ;
```